module/platform/webots/Webots/src/Webots.cpp

```diff
@@ -301,13 +306,29 @@ namespace module::platform::webots {
             // Service the watchdog
             emit<Scope::WATCHDOG>(ServiceWatchdog<Webots>());
 
-            // Tick the clock forward
-            Clock::tick();
+            // Deal with time
+
+            // Save our previous deltas
+            const uint32_t prev_sim_delta  = this->sim_delta;
+            const uint32_t prev_real_delta = this->real_delta;
+
+            // Update our current deltas
+            this->sim_delta  = msg.time - this->current_sim_time;
+            this->real_delta = msg.real_time - this->current_real_time;
+
+            // Calculate our custom rtf - the ratio of the past two sim deltas and the past two real time deltas
+            utility::clock::custom_rtf = static_cast<double>(this->sim_delta + prev_sim_delta)
+                                         / static_cast<double>(this->real_delta + prev_real_delta);
+
+            // Update our current times
+            this->current_sim_time  = msg.time;
+            this->current_real_time = msg.real_time;
```

Comment:
Do we really need all this `this->`?

Reply:
I think so, because I'm pretty sure the lamdas can't capture class variables. I think that if we want to use them, we have to use `this->var` or call a getter/setter on `this->get/set()` anyway

Reply:
```c++
// Type your code here, or load an example.
#include <iostream>
class A {
    int x = 1;

public:
    void lambda(){
        [this](){std::cout << x;}();
    }
};

int main(){
    A a;
    a.lambda();
}
```
This compiles and prints 1

Reply:
I mean, you can try without the `this`. When I tried with the `fd` one you also commented on, it said I couldn't capture it, and I found a stackoverflow thread which said yea you can't do that, so I used `this` here too 🤷

Reply:
You don't need the this if you capture this in the lambda

Reply:
Roger that. No idea why I had that issue before. Cameron's PR to this branch removes the this's #455 and multiplies by the `num`